crates/rethnet_eth/tests/integration_tests.rs

```diff
@@ -0,0 +1,361 @@
+use bytes::Bytes;
+
+use rethnet_eth::{
+    remote::{
+        eth::eip712,
+        methods::{FilterOptions, GetLogsInput, MethodInvocation, TransactionInput},
+        BlockSpec, BlockTag,
+    },
+    Address, B256, U256,
+};
+use rethnet_test_utils::help_test_method_invocation_serde;
+
+#[test]
+fn test_serde_eth_accounts() {
+    help_test_method_invocation_serde(MethodInvocation::Accounts());
+}
+
+#[test]
+fn test_serde_eth_block_number() {
+    help_test_method_invocation_serde(MethodInvocation::BlockNumber());
+}
+
+#[test]
+fn test_serde_eth_call() {
+    for block_spec in [
+        Some(BlockSpec::Number(U256::from(100))),
```

Comment:
I think we still would want the variety of `Some(...)` vs `None`, so I think since we're already committed to some variety we might as well just keep the `Number` vs `Tag` variety, but I'm happy to defer to your opinion there.

Reply:
I mean, there has to be an actual purpose to what we're testing. AFAICT, this test tests:

1. Primary: Selection of a default `BlockSpec` when `block_spec` is `None`. This is unique to `MethodInvocation::Call`, so has to be tested here
2. Secondary: (De)serialization of `TransactionInput`. Not unique to `MethodInvocation::Call`
3. Secondary: (De)serialization of some variants of `Some(BlockSpec)` but not all (e.g. pending, earliest). Not unique to `MethodInvocation::Call`

As such, having `None` and any one `Some` should suffice, as long as we have a dedicated test for ALL variants of `BlockSpec`.

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/4132/commits/39c66cc6533e33364d0176a6ec162d68226efe32